my/core/__main__.py

```diff
@@ -360,69 +329,121 @@ def module_install(args: Namespace) -> None:
     check_call(cmd)
 
 
-from argparse import ArgumentParser
-def parser() -> ArgumentParser:
-    p = ArgumentParser('Human Programming Interface', epilog='''
-Tool for HPI.
+@click.group()
+def main() -> None:
+    '''
+    Human Programming Interface
 
-Work in progress, will be used for config management, troubleshooting & introspection
-''')
-    sp = p.add_subparsers(dest='mode')
-    dp = sp.add_parser('doctor', help='Run various checks')
-    dp.add_argument('--verbose', action='store_true', help='Print more diagnosic infomration')
-    dp.add_argument('--all'    , action='store_true', help='List all modules, including disabled')
-    dp.add_argument('--quick'  , action='store_true', help='Only run partial checks (first 100 items)')
-    dp.add_argument('module', nargs='?', type=str   , help='Pass to check a specific module')
-    dp.set_defaults(func=doctor)
+    Tool for HPI
+    Work in progress, will be used for config management, troubleshooting & introspection
+    '''
+    # note: code in groups doesnt run unless a subcommand is passed
+    # https://click.palletsprojects.com/en/7.x/commands/#group-invocation-without-command
+    # https://click.palletsprojects.com/en/7.x/commands/#multi-command-chaining
 
-    cp = sp.add_parser('config', help='Work with configuration')
-    scp = cp.add_subparsers(dest='config_mode')
-    if True:
-        ccp = scp.add_parser('check', help='Check config')
-        ccp.set_defaults(func=config_check_cli)
+    # use a particular directory instead of a random one, since
```

Comment:
Ah fair enough -- and anyway being a `main` it basically means it's a context manager of sorts on its own, so chdir won't hurt. Maybe the only place where it can leave process in a different state is in tests?

Reply:
Yeah, I was considering adding a contextmangaer so it'd be possible to do that in the commands itself, when it was necessary, but I didn't know what it was needed for. Is it the `importlib` statements?

Reply:
Yeah, I think likely to avoid importing stuff by accident from cwd during development, with various sorts of automatic discoveries can be confusing. More of a paranoid thing I guess, maybe can remove later with more confidence/test coverage? Can be confusing too if some commands start producing outputs at some point, and people use relative paths

Reply:
Yeah, thats a good point. Typically the defaults for click are pretty good - the only time its tripped me up is tuples vs. lists